lua/neogit/lib/git/status.lua

```diff
@@ -130,13 +127,15 @@ local function update_status(state)
 
   state.cwd = cwd
   state.head = head
-  state.upstream = upstream
+  state.upstream.ref = upstream.ref
```

Comment:
I don't think this change is right. `state.upstream` needs to be set to the local `upstream` table. I think you can exercise this part by creating a new branch locally (it has no upstream yet)

Reply:
Right, I suspected such might be the case.

The reason with setting this as a table is that it will race with the `update_branch_information` async function and 50% of the time causes the abbrev to be overwritten by the new table

Reply:
I.e

![image](https://github.com/NeogitOrg/neogit/assets/25723553/6e624a2d-c16d-4110-ac20-af6e75bcd658)

These two will compete over who can set it last.

Reply:
Hm. Yeah, thats an issue. But the problem I've seen when just trying to update specific parts of it is that when creating a new branch, the `upstream` info persists to this new branch, and you wind up with neogit thinking the upstream is `origin/master` when thats plainly wrong.

There's probably some middle way that solves both issues. Selective nullification?